### PR TITLE
 generate docs for feature gated items

### PR DESCRIPTION
**Pull request by niklasad1**
I noticed that the stuff hidden behind the `unstable-light-client` doesn't show up on docs.rs/subxt which this PR fixes.

`docsrs` is still unstable and needs to enabled if one wants to use it, https://github.com/rust-lang/rust/issues/43781

### CODE REVIEW

File: codegen/Cargo.toml [hunk 1]
```diff
@@ -40,3 +40,10 @@ bitvec = { workspace = true }
 scale-info = { workspace = true, features = ["bit-vec"] }
 pretty_assertions = { workspace = true }
 frame-metadata = { workspace = true }
+
+[package.metadata.docs.rs]
+all-features = true
```
Comment (jsdw) [inline codegen/Cargo.toml:45]: Can we test this somehow? would `all-features = true` enable eg both `native` _and_ `web` and cause a compile error for instance, or does it avoid that?

File: subxt/src/tx/signer.rs [hunk 2]
```diff
@@ -25,6 +25,7 @@ pub trait Signer<T: Config> {
 }
 
 #[cfg(feature = "substrate-compat")]
+#[cfg_attr(docsrs, doc(cfg(feature = "substrate-compat")))]
```
Comment (jsdw) [inline subxt/src/tx/signer.rs:28]: This line that we have to add below each feature is sortof ugly! I'm not really familiar with it; what does it do? I also wonder whether we could avoid this line duplication somehow.

Comment (niklasad1) [reply subxt/src/tx/signer.rs:28]: we could write a macro for it as we have in jsonrpsee

```rust
macro_rules! cfg_feature {
	($feature:literal, $($item:item)*) => {
		$(
			#[cfg(feature = $feature)]
			#[cfg_attr(docsrs, doc(cfg(feature = $feature)))]
			$item
		)*
	}
}

```

File: codegen/Cargo.toml (hunk 1 above)
Comment (niklasad1) [reply codegen/Cargo.toml:45]: yeah, you are right I have to remove that

File: subxt/src/lib.rs [hunk 3]
```diff
@@ -55,6 +57,10 @@ pub mod storage;
 pub mod tx;
 pub mod utils;
 
+// Macros useful below, but not to be exposed outside of the crate.
```
Comment (niklasad1) [inline subxt/src/lib.rs:60]:
```suggestion
// Internal helper macros
```

**Review by tadeohepperle — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

**Pull request merged**